lib/exirc/utils.ex

```diff
@@ -25,7 +25,7 @@ defmodule ExIrc.Utils do
     end
   end
 
-  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>[\w.:-]+)))?$/
+  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>\S+[\w.:-]+)))?$/
```

Comment:
Actually, the nickname matching can be improved too, since it currently accepts whitespace, and you probably shouldn't accept tabs and newlines in usernames either :P

```elixir
  @prefix_pattern ~r/^(?<nick>[^!\s]+)(?:!(?<user>[^@\s]+@)?(?<host>\S+))?$/
```